### PR TITLE
[Liquidation bot] tests + scaling fixes

### PR DESCRIPTION
**Pull request by antonina-cherednichenko**
Add tests and more fixes to liquidation bot

### CODE REVIEW

File: test/liquidation/liquidation-bot-test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    const afterUSDCBalance = await usdc.balanceOf(owner.address);
+    const profit = afterUSDCBalance - beforeUSDCBalance;
+    expect(tx.hash).to.be.not.null;
+    expect(profit).to.be.greaterThan(0);
+  });
+
+  it('Should execute WETH flash swap with profit', async () => {
+    const { comet, liquidator, users: [owner, underwater], assets: { usdc, weth } } = await makeLiquidatableProtocol();
+    await weth.connect(underwater).approve(comet.address, 120000000000000000000n);
+    await comet.connect(underwater).supply(weth.address, 120000000000000000000n); //
```
Comment (scott-silver) [inline test/liquidation/liquidation-bot-test.ts:44]: nit: extra comment slashes `//`

File: contracts/liquidator/Liquidator.sol [hunk 2]
```diff
@@ -145,19 +174,19 @@ contract Liquidator is IUniswapV3FlashCallback, PeripheryImmutableState, Periphe
...
 
+            if (collateralBalance == 0) continue;
+
+            uint256 quotePrice = comet.quoteCollateral(asset, QUOTE_PRICE_SCALE * comet.baseScale());
+            uint256 assetBaseAmount = comet.baseScale() * QUOTE_PRICE_SCALE * collateralBalance / quotePrice;
             /*
-                quoteCollateral = amount of DAI you get for 1 USDC
-                collateralBalance = Comet's balance of DAI
-                price = amount of USDC required to buy all DAI
+                quoteCollateral = amount of asset you get for 1 USDC
```
[the remark this answers is not in the document]
Comment (antonina-cherednichenko) [reply contracts/liquidator/Liquidator.sol:183]: Good point, I will update these with other comments in separate pr

File: test/liquidation/liquidation-bot-test.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      reversedPair: false,
+    });
+
+    const afterUSDCBalance = await usdc.balanceOf(owner.address);
+    const profit = afterUSDCBalance - beforeUSDCBalance;
+    expect(tx.hash).to.be.not.null;
+    expect(profit).to.be.greaterThan(0);
+  });
+
+  it('Successful execution should increase base token balance of liquidating user', async () => {
```
[the remark this answers is not in the document]
Comment (antonina-cherednichenko) [reply test/liquidation/liquidation-bot-test.ts:141]: yes, this check is also in DAI main test, so this check became redundant.

File: contracts/liquidator/Liquidator.sol [hunk 4]
```diff
@@ -34,30 +34,38 @@ contract Liquidator is IUniswapV3FlashCallback, PeripheryImmutableState, Periphe
     using LowGasSafeMath for uint256;
     using LowGasSafeMath for int256;
 
+    uint256 public constant QUOTE_PRICE_SCALE = 1e6;
```
[the remark this answers is not in the document]
Comment (antonina-cherednichenko) [reply contracts/liquidator/Liquidator.sol:37]: This one is tricky. It's more like how much of the asset I can buy for 1e6 * 1e6 USDC. For WBTC we were losing precision when asking how much of the asset we can buy per 1e6 USDC. 

The formula is subject to improvement as well.

**Pull request merged**